content/react/concepts/react-native/react-native-components/react-native-components.md

```diff
@@ -0,0 +1,160 @@
+
+## React Native Components are based on React
+
+![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
+
+React Native Components share the same API structure as React Components. Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as Function Component or Class Components:
+
+### Function Component:
+
+```jsx
+import React from 'react';
+import { Text } from 'react-native';
+
+const Box= () => {
+    return (
+        <Text>I have a small box</Text>
+    );
+}
+
+export default Box;
+```
+    
+### Class Component:
+
+```jsx
+import React, { Component } from "react";
+import { Text } from 'react-native';
+
+class Box extends Component {
+    render() {
+        return (
+            <Text>I have a small box</Text>
+        );
+    }
+}
+
+export default Box;
+```
+	
+## JSX, Props and State.
+
+React Native Components also use JSX, accept Props and manage State.
+
+### JSX
+
+As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
+
+```jsx
+import React from 'react';
+import { Text } from 'react-native';
+
+const Box= () => {
+    const size = “small”;
+    return (	
+        <Text>I have a {size} box</Text>
+    );
+}
+
+export default Box;
+```
+	
+### Props
+
+Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:
+
+```jsx
+import React from 'react';
+import { View, Text } from 'react-native';
+
+const Box = (props) => {
+    return (
+        <Text>I have a {props.size} box</Text>
+    );
+}
+
+const BoxCollection = () => {
+    return (
+        <View>
+            <Box size=“small” />
+            <Box size=“medium” />
+            <Box size=“large” />
+        </View>
+    );
+}
+
+export default BoxCollection;
+```
+	
+### State
+
+Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:
+
+```jsx
+import React, { useState } from "react";
+import { View, Text, Button } from "react-native";
+
+const Box = () => {
+    const [size, setSize] = useState("small");
+        return (
+            <View>
+                <Text>I have a {size} box </Text>
+                <Button
+                    color="red"
+                    onPress={() => setSize("small")}
+                    title="Small" />
+                <Button
+                    color="blue"
+                    onPress={() => setSize("medium")}
+                    title="Medium" />
+                <Button
+                    color="orange"
+                    onPress={() => setSize("large")}
+                    title="Large" />
+            </View>
+        );
+}
+
+export default Box;
```

Comment:
```suggestion
import React, { useState } from "react";
import { View, Text, Button } from "react-native";

const Box = () => {
  const [size, setSize] = useState("small");
  return (
    <View>
      <Text>I have a {size} box </Text>
      <Button
        color="red"
        onPress={() => setSize("small")}
        title="Small" />
      <Button
        color="blue"
        onPress={() => setSize("medium")}
        title="Medium" />
      <Button
        color="orange"
        onPress={() => setSize("large")}
        title="Large" />
    </View>
  );
}

export default Box;
```

Reply:
Hey @Dusch4593 thanks for the review! I've made the changes but before I update this pull request, I have a question if you don't mind. 

I'm new to Git and Github, and I'm confused by something. For some reason, I don't know if I did something wrong or what, but some files that I didn't created nor modified started showing up when staging files to my forked repository of the Codecademy docs. 

What should I do with this files? I believe if I commit the changes to this React Native entry, these files would appear from my branch and would be merged with this entry, I don't know if those files are up to date. Should I delete these? then commit and update the pull request?

<img width="777" alt="Screenshot 2022-12-18 at 7 43 58 PM" src="https://user-images.githubusercontent.com/95394815/208325986-ed5ef6d1-6170-4e4e-900a-3bf2ac340b1b.png">

Reply:
Thanks @Dusch4593 I did that without the --hard flag and It says branch is now updated. This is what I get know. I assume all these files are ignored when I commit rn_components right?

<img width="1054" alt="Screenshot 2022-12-21 at 12 50 27 AM" src="https://user-images.githubusercontent.com/95394815/208826240-bbd1dffa-f261-436d-8e00-b52b61694cae.png">

Reply:
@ericsonrd Cool! Now, can we run `git checkout main` and run `git status` and see what the screenshot shows there?

Reply:
@Dusch4593 Before checking out to main, it prompts me to remove these: 
<img width="864" alt="Screenshot 2022-12-26 at 1 07 02 PM" src="https://user-images.githubusercontent.com/95394815/209570475-89c1eb1b-bb30-4dd6-a376-c317279a6ef0.png">

Reply:
@ericsonrd Try `git stash` and then switch over to the main branch. If I'm trying to maneuver quickly between branches and if the current branch I'm working on has changes on it that I don't want to commit just yet.

Reply:
HI @yangc95, thanks for answering. I did that before but the group of files at the bottom still would be overwirtten. Can these be safely removed then checkout?
<img width="889" alt="Screenshot 2022-12-26 at 11 58 14 PM" src="https://user-images.githubusercontent.com/95394815/209608838-58b3ed86-97cd-444b-adc2-bbe1447aeca3.png">

Reply:
@ericsonrd You could try `git stash --include-untracked` and then you should be able to checkout into `main`.

Also, if you want to see the running list of changes you've stashed, run `git stash list`. 😄

Reply:
@Dusch4593 Done. It's checked out to `main`. Now what's next? Update `main' from the Codecademy docs then Merge `main` into `rn_components` and continue from there?

<img width="891" alt="Screenshot 2022-12-27 at 8 11 05 PM" src="https://user-images.githubusercontent.com/95394815/209738458-514c8c07-c526-402b-b15c-706c8023e950.png">

Reply:
@ericsonrd Awesome! Now even there it says that remote `main` is in sync with `origin/main`, what happens if we run `git pull`?

Reply:
@Dusch4593 Says it's already up to date

Reply:
Gotcha. Just wanted to confirm your local `main` was at least untangled. Now, it looks like your remote `origin/main` may need untangled next:

<img width="903" alt="Screen Shot 2022-12-28 at 6 27 38 PM" src="https://user-images.githubusercontent.com/23144845/209884815-f166256b-538d-44f0-916f-2f8db66fb79f.png">

It is expected for your fork's `origin/main` is x-commits _behind_ the base `Codecademy/docs:main` branch since new changes from outside PR branches are being merged. But sync problems may occur for your fork's `origin/main` if it is x-commits _ahead_, as shown above. A solution that's worked for me is a hard `reset` back to the base `main` branch. On your local `main` branch, assuming you have already have a remote named `upstream` pointing to `https://github.com/Codecademy/docs.git`, let's run the following commands:

```
git reset --hard upstream/main
```

This solution was borrowed from [this StackOverflow thread](https://stackoverflow.com/questions/41283955/github-keeps-saying-this-branch-is-x-commits-ahead-y-commits-behind#:~:text=Now%20to%20fix,match%20upstream%3A)

Give this a try and let me know if you have any issues. 😄

Reply:
Done. This is what is shows:

<img width="562" alt="Screenshot 2022-12-28 at 7 56 50 PM" src="https://user-images.githubusercontent.com/95394815/209886566-cdeb3f03-1061-486a-98af-6a355f30e127.png">

Reply:
Ok now try `git pull` again.

Reply:
Done. 

<img width="1112" alt="Screenshot 2022-12-28 at 8 11 47 PM" src="https://user-images.githubusercontent.com/95394815/209887330-e15dc99e-4b70-4eec-9a5b-43e0f5257943.png">
<img width="697" alt="Screenshot 2022-12-28 at 8 12 02 PM" src="https://user-images.githubusercontent.com/95394815/209887338-16379560-66e5-4129-921f-408ee45b46b4.png">

Reply:
Hey @ericsonrd! Hoping you had a restful holiday and happy new year! 🎆

I'm sorry that [your fork's `main` branch](https://github.com/ericsonrd/cc-docs-frk/tree/main) still seems to be a bit entangled. After discussing this with a colleague, we were wondering if you'd be open to lifting your fork's branch protections for today to see if we can fix this on our end?

cc: @yangc95

Reply:
Hi @Dusch4593! Happy new year to you too!

Done with the branch protections, let me know if it works. I've been reading some guides and now everything is up to date except for the "commits ahead" on main/rn_components, so it seems the next step is Rebase, right?

Reply:
@Dusch4593 @ericsonrd it looks like permissions are still denied. Let's continue this conversation in https://github.com/Codecademy/docs/discussions/categories/q-a